bundles/binding/org.openhab.binding.gc100ir/src/main/java/org/openhab/binding/gc100ir/internal/GC100IRActiveBinding.java

```diff
@@ -0,0 +1,338 @@
+	 * 
+	 * @param itemName
+	 * @return a String value of GC100 instance.
+	 */
+	private String getGC100Instance(String itemName) {
+		for (BindingProvider provider : providers) {
+			if (provider instanceof GC100IRBindingProvider) {
+				GC100IRBindingProvider gc100Provider = (GC100IRBindingProvider) provider;
+				if (gc100Provider.getItemNames().contains(itemName)) {
+					return gc100Provider.getGC100Instance(itemName);
+				}
+			}
+		}
+		return null;
+	}
+
+	/**
+	 * Gets the GC100 connector object from connector list.
+	 * 
+	 * @param gc100Instance
+	 * @return GC100IRConnector instance.
+	 */
+	private GC100IRConnector getGC100Connector(String gc100Instance) {
+		// sanity check
+		if (gc100Instance == null)
+			return null;
+
+		// check if the connector for this instance already exists
+		GC100IRConnector connector = connectors.get(gc100Instance);
+		if (connector != null)
+			return connector;
+
+		GC100IRHost gc100Host;
+		if (gc100Instance.startsWith("#")) {
+			// trim off the '#' identifier
+			String instance = gc100Instance.substring(1);
+
+			// check if we have been initialised yet - can't process
+			// named instances until we have read the binding config
+			if (nameHostMapper == null) {
+				logger.trace(
+						"Attempting to access the named instance '{}' before the binding config has been loaded",
+						instance);
+				return null;
+			}
+
+			// check this instance name exists in our config
+			if (!nameHostMapper.containsKey(instance)) {
+				logger.error(
+						"Named instance '{}' does not exist in the binding config",
+						instance);
+				return null;
+			}
+
+			gc100Host = nameHostMapper.get(instance);
+		} else {
+			gc100Host = new GC100IRHost();
+			gc100Host.setHostname(gc100Instance);
+		}
+
+		// create a new connection handler
+		logger.debug("Creating new GC100IRConnector for '{}' on {}",
+				gc100Instance, gc100Host.getHostname());
+		connector = new GC100IRConnector(gc100Host, eventPublisher);
+		connectors.put(gc100Instance, connector);
+
+		// attempt to open the connection straight away
+		try {
+			connector.open();
+		} catch (Exception e) {
+			logger.error("Connection failed for '{}' on {}", gc100Instance,
+					gc100Host.getHostname());
+		}
+
+		return connector;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		for (Map.Entry<String, GC100IRConnector> entry : connectors.entrySet()) {
+			GC100IRConnector connector = entry.getValue();
+			if (connector.isConnected()) {
+				// we are still connected but send a ping to make sure
+				if (connector.ping())
+					continue;
+
+				// broken connection so attempt to reconnect
+				logger.debug(
+						"Broken connection found for '{}', attempting to reconnect...",
+						entry.getKey());
+				try {
+					connector.open();
+				} catch (Exception e) {
+					logger.debug(
+							"Reconnect failed for '{}', will retry in {}s",
+							entry.getKey(), refreshInterval / 1000);
+				}
+			}
+		}
```

Comment:
you seen to use the excute() method to issue a heartbeat functionality which is not quite intended for it. You should therefore explain the purpose of your implementation in more detail